### PR TITLE
Gum.Kni

### PR DESCRIPTION
**Pull request by nkast**

### CODE REVIEW

File: ToolsUtilities/FileManager.cs [hunk 1]
```diff
@@ -21,16 +21,29 @@ public static partial class FileManager
...
+                try
+                {
+                    return Path.GetDirectoryName(AppContext.BaseDirectory)
+                            .Replace('/', Path.DirectorySeparatorChar)
+                            .Replace('\\', Path.DirectorySeparatorChar) + Path.DirectorySeparatorChar;
+                }
+                catch 
+                {
+                    // WASM ?
+                    return "./";
```
Comment (nkast) [inline ToolsUtilities/FileManager.cs:42]: AppContext.BaseDirectory throws in WASM, and I changed ExeLocation to "./".
I am not sure if that is the correct fix here.

**Comment by nkast** [comment]
DesktopGL & WindowsDX seems to work.
BlazorGL will run without exceptions, but tries to load content from `http://localhost:5259/Content/Content/BearTexture.png`
and sprites/textures will not show. The correct address of the file is 'http://localhost:5259/Content/BearTexture.png'

File: MonoGameGum/KniGum.csproj [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    <PackageId>Gum.KNI</PackageId>
+    <Version>2024.6.20.2</Version>
+
+    <GenerateDocumentationFile>true</GenerateDocumentationFile>
+
+
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
+    <DefineConstants>$(DefineConstants);MONOGAME;KNI;USE_GUMCOMMON</DefineConstants>
```
Comment (nkast) [inline MonoGameGum/KniGum.csproj:17]: Here I define both MONOGAME and KNI. Perhaps it's best to define only KNI and expand all 
`#if MONOGAME` 
to
`#if MONOGAME || KNI`

File: ToolsUtilities/FileManager.cs (hunk 1 above)
Comment (vchelaru) [reply ToolsUtilities/FileManager.cs:42]: I think that's okay, unless we can get a proper #if

File: MonoGameGum/KniGum.csproj (hunk 2 above)
Comment (vchelaru) [reply MonoGameGum/KniGum.csproj:17]: Ya I think that's probably best, even though it's more work.

File: ToolsUtilities/FileManager.cs (hunk 1 above)
Comment (nkast) [reply ToolsUtilities/FileManager.cs]: Actually it was Path.GetDirectoryName() that was failing.
Now I skip it when the value is "/", and let the rest of the method to do, whatever it does.
This also fixed the issue we had with /Content/Content/ path!

File: MonoGameGum/KniGum.csproj (hunk 2 above)
Comment (nkast) [reply MonoGameGum/KniGum.csproj]: Done.

**Comment by nkast** [comment]
KniGumInCode

Here is the latest KniGumInCode on the web.
The issue with /Content/Content/ disappeared once we have the proper fix for ExeLocation.

**Comment by nkast** [comment]
KniGumFromFile

Here is the latest KniGumFromFile on WindowsDX.

File: GumDataTypes/GumProjectSave.cs [hunk 3]
```diff
@@ -422,21 +422,24 @@ private void PopulateElementSavesFromReferences(string projectRootDirectory, Lin
                 }
             }
 
-            foreach(var reference in BehaviorReferences)
+            if  (BehaviorReferences != null)
```
Comment (nkast) [inline GumDataTypes/GumProjectSave.cs:425]: I tried rooting the Gum assemblies from the main project, and setting IsTrimmable on the libraries.
That seems to be XmlSerializer behavior.

**Pull request merged**